py-polars/polars/internals/expr.py

```diff
@@ -1794,6 +1794,20 @@ def diff(self, n: int = 1, null_behavior: str = "ignore") -> "Expr":
         """
         return wrap_expr(self._pyexpr.diff(n, null_behavior))
 
+    def pct_change(self, n: int = 1) -> "Expr":
+        """
+        Percentage change (as fraction) between current element and most-recent
+        non-null element n or more periods before.
```

Comment:
Could you be a bit more specific here (and same docstring for Series)? I am confused by the statement "most-recent non-null element n or more periods before". The "most-recent" + "n or more" suggests you would consider multiple values for `n>1`, but if `n==2`, you would *always*  compare with two elements before. If one of the two is null, the output would be null. Well, that is what Pandas does, and what I would expect to happen.

```python
s = pl.Series([1, 3, None, 4])
s.pct_change(n=2)
```

Reply:
@zundertj it does work the same as Pandas `pct_change`. I tried to make the doc-string a little bit more specific by saying "at least n period(s) before" instead of the "n or more".